### ISSUE TITLE
[hydrated_bloc] example crashes on web (Chrome macOS)

### ISSUE THREAD

**Issue by nosmirck** [felangel/bloc#3363]
**Description**
Running the hydrated_bloc example crashes on Chrome on macOS.

**Steps To Reproduce**
run the example project on Chrome on macOS

**Expected Behavior**
Should not crash

**Console Log***
```
Launching lib/main.dart on Chrome in debug mode...
This app is linked to the debug service: ws://127.0.0.1:57966/0b2ZGpoSWI0=/ws
Debug service listening on ws://127.0.0.1:57966/0b2ZGpoSWI0=/ws
💪 Running with sound null safety 💪
Connecting to VM Service at ws://127.0.0.1:57966/0b2ZGpoSWI0=/ws
Flutter Web Bootstrap: Programmatic
Error: Unexpected null value.
    at Object.throw_ [as throw] (http://localhost:57916/dart_sdk.js:5080:11)
    at Object.nullCheck (http://localhost:57916/dart_sdk.js:5399:30)
at hive_impl.HiveImpl.new._openBox (http://localhost:57916/packages/hive/src/box_collection.dart.lib.js:3816:37)
    at _openBox.next (<anonymous>)
    at runBody (http://localhost:57916/dart_sdk.js:40660:34)
    at Object._async [as async] (http://localhost:57916/dart_sdk.js:40691:7)
at hive_impl.HiveImpl.new.[_openBox] (http://localhost:57916/packages/hive/src/box_collection.dart.lib.js:3788:20)
at hive_impl.HiveImpl.new.openBox (http://localhost:57916/packages/hive/src/box_collection.dart.lib.js:3856:52)
    at openBox.next (<anonymous>)
    at runBody (http://localhost:57916/dart_sdk.js:40660:34)
    at Object._async [as async] (http://localhost:57916/dart_sdk.js:40691:7)
at hive_impl.HiveImpl.new.openBox (http://localhost:57916/packages/hive/src/box_collection.dart.lib.js:3852:20)
at http://localhost:57916/packages/hydrated_bloc/src/hydrated_storage.dart.lib.js:50:62
    at Generator.next (<anonymous>)
    at runBody (http://localhost:57916/dart_sdk.js:40660:34)
    at Object._async [as async] (http://localhost:57916/dart_sdk.js:40691:7)
at http://localhost:57916/packages/hydrated_bloc/src/hydrated_storage.dart.lib.js:45:120
at basic_lock.BasicLock.new.synchronized (http://localhost:57916/packages/synchronized/src/reentrant_lock.dart.lib.js:134:24)
    at synchronized.next (<anonymous>)
    at runBody (http://localhost:57916/dart_sdk.js:40660:34)
    at Object._async [as async] (http://localhost:57916/dart_sdk.js:40691:7)
at basic_lock.BasicLock.new.synchronized (http://localhost:57916/packages/synchronized/src/reentrant_lock.dart.lib.js:122:20)
at Function.build (http://localhost:57916/packages/hydrated_bloc/src/hydrated_storage.dart.lib.js:45:53)
at http://localhost:57916/packages/hydrated_bloc_test/main.dart.lib.js:676:51
    at Generator.next (<anonymous>)
    at runBody (http://localhost:57916/dart_sdk.js:40660:34)
    at Object._async [as async] (http://localhost:57916/dart_sdk.js:40691:7)
at http://localhost:57916/packages/hydrated_bloc_test/main.dart.lib.js:674:223
at hydrated_bloc._HydratedBlocOverridesScope.new._init (http://localhost:57916/packages/hydrated_bloc/src/hydrated_bloc.dart.lib.js:176:84)
    at _init.next (<anonymous>)
    at runBody (http://localhost:57916/dart_sdk.js:40660:34)
    at Object._async [as async] (http://localhost:57916/dart_sdk.js:40691:7)
at hydrated_bloc._HydratedBlocOverridesScope.new.[_init] (http://localhost:57916/packages/hydrated_bloc/src/hydrated_bloc.dart.lib.js:174:20)
at http://localhost:57916/packages/hydrated_bloc/src/hydrated_bloc.dart.lib.js:114:31
    at Generator.next (<anonymous>)
    at runBody (http://localhost:57916/dart_sdk.js:40660:34)
    at Object._async [as async] (http://localhost:57916/dart_sdk.js:40691:7)
at http://localhost:57916/packages/hydrated_bloc/src/hydrated_bloc.dart.lib.js:113:145
    at _rootRun (http://localhost:57916/dart_sdk.js:40931:14)
    at async._CustomZone.new.run (http://localhost:57916/dart_sdk.js:40178:14)
    at Object._runZoned (http://localhost:57916/dart_sdk.js:41078:92)
    at Object.runZoned (http://localhost:57916/dart_sdk.js:41036:18)
at http://localhost:57916/packages/hydrated_bloc/src/hydrated_bloc.dart.lib.js:113:90
    at _rootRun (http://localhost:57916/dart_sdk.js:40931:14)
    at async._CustomZone.new.run (http://localhost:57916/dart_sdk.js:40178:14)
    at Object._runZoned (http://localhost:57916/dart_sdk.js:41078:92)
    at Object.runZoned (http://localhost:57916/dart_sdk.js:41036:18)
at Function.runZoned (http://localhost:57916/packages/bloc/src/transition.dart.lib.js:578:19)
at Function.runZoned (http://localhost:57916/packages/hydrated_bloc/src/hydrated_bloc.dart.lib.js:113:33)
at main$ (http://localhost:57916/packages/hydrated_bloc_test/main.dart.lib.js:674:43)
    at main$.next (<anonymous>)
    at runBody (http://localhost:57916/dart_sdk.js:40660:34)
    at Object._async [as async] (http://localhost:57916/dart_sdk.js:40691:7)
at main$ (http://localhost:57916/packages/hydrated_bloc_test/main.dart.lib.js:673:18)
at http://localhost:57916/web_entrypoint.dart.lib.js:34:33
    at Object._checkAndCall (http://localhost:57916/dart_sdk.js:5279:16)
    at Object.dcall (http://localhost:57916/dart_sdk.js:5284:17)
    at http://localhost:57916/dart_sdk.js:140291:18
    at Generator.next (<anonymous>)
    at http://localhost:57916/dart_sdk.js:40641:33
    at _RootZone.runUnary (http://localhost:57916/dart_sdk.js:40511:59)
    at _FutureListener.thenAwait.handleValue (http://localhost:57916/dart_sdk.js:35438:29)
    at handleValueCallback (http://localhost:57916/dart_sdk.js:35999:49)
    at Function._propagateToListeners (http://localhost:57916/dart_sdk.js:36037:17)
    at _Future.new.[_completeWithValue] (http://localhost:57916/dart_sdk.js:35872:23)
    at async._AsyncCallbackEntry.new.callback (http://localhost:57916/dart_sdk.js:35906:35)
    at Object._microtaskLoop (http://localhost:57916/dart_sdk.js:40778:13)
    at _startMicrotaskLoop (http://localhost:57916/dart_sdk.js:40784:13)
    at http://localhost:57916/dart_sdk.js:36261:9
Error: Unexpected null value.
    at Object.throw_ [as throw] (http://localhost:57916/dart_sdk.js:5080:11)
    at Object.nullCheck (http://localhost:57916/dart_sdk.js:5399:30)
at hive_impl.HiveImpl.new._openBox (http://localhost:57916/packages/hive/src/box_collection.dart.lib.js:3816:37)
    at _openBox.next (<anonymous>)
    at runBody (http://localhost:57916/dart_sdk.js:40660:34)
    at Object._async [as async] (http://localhost:57916/dart_sdk.js:40691:7)
at hive_impl.HiveImpl.new.[_openBox] (http://localhost:57916/packages/hive/src/box_collection.dart.lib.js:3788:20)
at hive_impl.HiveImpl.new.openBox (http://localhost:57916/packages/hive/src/box_collection.dart.lib.js:3856:52)
    at openBox.next (<anonymous>)
    at runBody (http://localhost:57916/dart_sdk.js:40660:34)
    at Object._async [as async] (http://localhost:57916/dart_sdk.js:40691:7)
at hive_impl.HiveImpl.new.openBox (http://localhost:57916/packages/hive/src/box_collection.dart.lib.js:3852:20)
at http://localhost:57916/packages/hydrated_bloc/src/hydrated_storage.dart.lib.js:50:62
    at Generator.next (<anonymous>)
    at runBody (http://localhost:57916/dart_sdk.js:40660:34)
    at Object._async [as async] (http://localhost:57916/dart_sdk.js:40691:7)
at http://localhost:57916/packages/hydrated_bloc/src/hydrated_storage.dart.lib.js:45:120
at basic_lock.BasicLock.new.synchronized (http://localhost:57916/packages/synchronized/src/reentrant_lock.dart.lib.js:134:24)
    at synchronized.next (<anonymous>)
    at runBody (http://localhost:57916/dart_sdk.js:40660:34)
    at Object._async [as async] (http://localhost:57916/dart_sdk.js:40691:7)
at basic_lock.BasicLock.new.synchronized (http://localhost:57916/packages/synchronized/src/reentrant_lock.dart.lib.js:122:20)
at Function.build (http://localhost:57916/packages/hydrated_bloc/src/hydrated_storage.dart.lib.js:45:53)
at http://localhost:57916/packages/hydrated_bloc_test/main.dart.lib.js:676:51
    at Generator.next (<anonymous>)
    at runBody (http://localhost:57916/dart_sdk.js:40660:34)
    at Object._async [as async] (http://localhost:57916/dart_sdk.js:40691:7)
at http://localhost:57916/packages/hydrated_bloc_test/main.dart.lib.js:674:223
at hydrated_bloc._HydratedBlocOverridesScope.new._init (http://localhost:57916/packages/hydrated_bloc/src/hydrated_bloc.dart.lib.js:176:84)
    at _init.next (<anonymous>)
    at runBody (http://localhost:57916/dart_sdk.js:40660:34)
    at Object._async [as async] (http://localhost:57916/dart_sdk.js:40691:7)
at hydrated_bloc._HydratedBlocOverridesScope.new.[_init] (http://localhost:57916/packages/hydrated_bloc/src/hydrated_bloc.dart.lib.js:174:20)
at http://localhost:57916/packages/hydrated_bloc/src/hydrated_bloc.dart.lib.js:114:31
    at Generator.next (<anonymous>)
    at runBody (http://localhost:57916/dart_sdk.js:40660:34)
    at Object._async [as async] (http://localhost:57916/dart_sdk.js:40691:7)
at http://localhost:57916/packages/hydrated_bloc/src/hydrated_bloc.dart.lib.js:113:145
    at _rootRun (http://localhost:57916/dart_sdk.js:40931:14)
    at async._CustomZone.new.run (http://localhost:57916/dart_sdk.js:40178:14)
    at Object._runZoned (http://localhost:57916/dart_sdk.js:41078:92)
    at Object.runZoned (http://localhost:57916/dart_sdk.js:41036:18)
at http://localhost:57916/packages/hydrated_bloc/src/hydrated_bloc.dart.lib.js:113:90
    at _rootRun (http://localhost:57916/dart_sdk.js:40931:14)
    at async._CustomZone.new.run (http://localhost:57916/dart_sdk.js:40178:14)
    at Object._runZoned (http://localhost:57916/dart_sdk.js:41078:92)
    at Object.runZoned (http://localhost:57916/dart_sdk.js:41036:18)
at Function.runZoned (http://localhost:57916/packages/bloc/src/transition.dart.lib.js:578:19)
at Function.runZoned (http://localhost:57916/packages/hydrated_bloc/src/hydrated_bloc.dart.lib.js:113:33)
at main$ (http://localhost:57916/packages/hydrated_bloc_test/main.dart.lib.js:674:43)
    at main$.next (<anonymous>)
    at runBody (http://localhost:57916/dart_sdk.js:40660:34)
    at Object._async [as async] (http://localhost:57916/dart_sdk.js:40691:7)
at main$ (http://localhost:57916/packages/hydrated_bloc_test/main.dart.lib.js:673:18)
at http://localhost:57916/web_entrypoint.dart.lib.js:34:33
    at Object._checkAndCall (http://localhost:57916/dart_sdk.js:5279:16)
    at Object.dcall (http://localhost:57916/dart_sdk.js:5284:17)
    at http://localhost:57916/dart_sdk.js:140291:18
    at Generator.next (<anonymous>)
    at http://localhost:57916/dart_sdk.js:40641:33
    at _RootZone.runUnary (http://localhost:57916/dart_sdk.js:40511:59)
    at _FutureListener.thenAwait.handleValue (http://localhost:57916/dart_sdk.js:35438:29)
    at handleValueCallback (http://localhost:57916/dart_sdk.js:35999:49)
    at Function._propagateToListeners (http://localhost:57916/dart_sdk.js:36037:17)
    at _Future.new.[_completeWithValue] (http://localhost:57916/dart_sdk.js:35872:23)
    at async._AsyncCallbackEntry.new.callback (http://localhost:57916/dart_sdk.js:35906:35)
    at Object._microtaskLoop (http://localhost:57916/dart_sdk.js:40778:13)
    at _startMicrotaskLoop (http://localhost:57916/dart_sdk.js:40784:13)
    at http://localhost:57916/dart_sdk.js:36261:9
```

**Additional Context**
***Chrome Version:***
Version 101.0.4951.64 (Official Build) (x86_64)

***Device Details:***
- macOS Monterey
- Version: 12.3.1 (21E258)
- MacBook Pro (15-inch, 2017)
- Processor: 2.9GHz Quad-Core Intel Core i7
- Memory 16GB 2133 MHz LPDDR3
- Graphics: Radeon Pro 560 4 GB Intel HD Graphics 630 1536 MB

***Flutter Doctor:***
```
[✓] Flutter (Channel stable, 3.0.0, on macOS 12.3.1 21E258 darwin-x64, locale en)
    • Flutter version 3.0.0 at /Users/lpulido/Library/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ee4e09cce0 (8 days ago), 2022-05-09 16:45:18 -0700
    • Engine revision d1b9a6938a
    • Dart version 2.17.0
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/lpulido/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/lpulido/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7590822)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7590822)

[✓] VS Code (version 1.67.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.40.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 12.3.1 21E258 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 101.0.4951.64

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

**Comment by nosmirck** [comment]
I've tested on Chrome with Windows PC and this issue is not reproducible. I'm starting to suspect it might be an issue with my computer, so, if anyone with a Mac (Intel) can try this and report their results I'd be grateful.

This was working previously to my update to Flutter 3

**Comment by nosmirck** [comment]
Well, it looks like I had my hive dependency updated to 2.2.0.

according to [this issue ](https://github.com/hivedb/hive/issues/967) on Hive's repo, it is a broken change on the latest version. I went and rolled hive back to a fixed 2.1.0 version until a fix is applied.

I'm closing this issue.
